### PR TITLE
fixed row totals bug #599

### LINKED ISSUE #599: row totals do not display correct info if no network connection
image

Should really be `Total: 0 Systems` - same issue occurs on every table

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

Fixed issue where table row counts not rendering properly with slow internet connection
Abstracted logic into function in `utils.tsx`

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking

closes #599

### CODE REVIEW

File: src/utils.tsx [hunk 1]
```diff
@@ -159,3 +159,14 @@ export const resetUniqueIdCounter = () => {
 export function sortDataList(data: any[], sortedValue: string) {
   return data.sort((a, b) => a[sortedValue].localeCompare(b[sortedValue]));
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (joelvdavies) [inline src/utils.tsx:163]:
```suggestion
```

File: src/utils.tsx
Comment (joelvdavies) [inline src/utils.tsx:1]: I think `getTableCountText` in `systems.component.tsx` should be replaced by this too.

File: src/systems/systems.component.tsx [hunk 2]
```diff
@@ -155,17 +159,17 @@ const CopySystemsButton = (props: {
...
+//   numRowsDisplayed: number,
+//   numSystems: number,
+//   isRoot: boolean
+// ) => {
+//   const systemsText = isRoot ? 'Systems' : 'Subsystems';
+//   if (numRowsDisplayed === numSystems)
+//     return `Total ${systemsText}: ${numSystems}`;
+//   else
+//     return `Returned ${numRowsDisplayed} out of ${numSystems} ${systemsText}`;
+// };
```
Comment (joelvdavies) [inline src/systems/systems.component.tsx:172]:
```suggestion
```

**Comment by joelvdavies** [comment]
@joshuadkitenge  Just noticed, should the catalogue category card view also have the same? It currently displays the full total, rather than changing based on the pagination like the tables.

**Review by joelvdavies — DISMISSED** [review]

**Review by joelvdavies — APPROVED** [review]
LGTM

**Review by joshuadkitenge — APPROVED** [review]
LGTM

**Pull request merged**